blocks/footer/footer.js

```diff
@@ -39,122 +39,136 @@ export default async function decorate(block) {
+      modal = await createModal([content]);
+      modal.showModal();
+    };
+
+    // Rendering the Store Switcher Modal Content
+    const $storeSwitcherBtn = footer.querySelector(
+      '.storeview-switcher-button',
+    );
+
+    // Store Switcher Modal Content
+    const storeSwitcherPath = '/store-switcher';
+    let fragmentStoreView;
+
+    try {
+      fragmentStoreView = await loadFragment(storeSwitcherPath);
+      if (!fragmentStoreView) throw new Error('Store Switcher fragment not found');
+    } catch (error) {
+      console.error('Error loading store switcher fragment:', error);
+      return;
+    }
+
+    // Store Switcher Modal Content
+    const storeSwitcher = document.createElement('div');
+
+    // Return Storename from stores-switcher
+    const selected = [...fragmentStoreView.querySelectorAll('a')].find((a) => {
+      const url = new URL(a.href);
+      return url.pathname.startsWith(root);
+    });
 
-  // create classes for storeview modal sections
-  const classes = ['storeview-title', 'storeview-list'];
-  classes.forEach((c, i) => {
-    const section = storeSwitcher.children[i];
-    if (section) section.classList.add(`storeview-modal-${c}`);
-  });
-
-  // Store Switcher Modal Content - Store View Title
-  const storeViewTitle = storeSwitcher.querySelector('.storeview-modal-storeview-title');
-  const title = storeViewTitle.querySelector('h3');
-  if (title) {
-    title.className = '';
-    title.closest('h3').classList.add('storeview-modal-storeview-title');
-    title.setAttribute('tabindex', '0');
-  }
+    storeSwitcher.id = 'storeview-modal';
+    while (fragmentStoreView.firstElementChild) {
+      storeSwitcher.append(fragmentStoreView.firstElementChild);
+    }
 
-  // Storeview List
-  const storeViewList = storeSwitcher.querySelector('.storeview-modal-storeview-list');
+    // create classes for storeview modal sections
+    const classes = ['storeview-title', 'storeview-list'];
+    classes.forEach((c, i) => {
+      const section = storeSwitcher.children[i];
+      if (section) section.classList.add(`storeview-modal-${c}`);
+    });
 
-  if (storeViewList) {
-    // Add storeview-selection class to parent UL
-    storeViewList
-      .querySelectorAll(':scope .default-content-wrapper > ul')
-      .forEach((storeView) => {
-        if (storeView.querySelector('ul')) storeView.classList.add('storeview-selection');
-      });
+    // Store Switcher Modal Content - Store View Title
+    const storeViewTitle = storeSwitcher.querySelector('.storeview-modal-storeview-title');
+    const title = storeViewTitle.querySelector('h3');
+    if (title) {
+      title.className = '';
+      title.closest('h3').classList.add('storeview-modal-storeview-title');
+      title.setAttribute('tabindex', '0');
+    }
+
+    // Storeview List
+    const storeViewList = storeSwitcher.querySelector('.storeview-modal-storeview-list');
+
+    if (storeViewList && storeViewList.children.length) {
```

Comment:
discussed and shown in standup: Added conditional: Only show Store Selection if /storeswitcher has children (aka nested links). If the storeswithcer document is blank or just contains the title, do not show storeSwitcher.